packages/shared/src/components/widgets/SocialShare.tsx

```diff
@@ -78,77 +98,129 @@ export const SocialShare = ({
-              onClick={() => onShareToSquad(squad)}
-            >
-              <SourceProfilePicture source={squad} />
-              <ShareText className="mt-2 break-words">
-                @{squad.handle}
-              </ShareText>
-            </button>
-          ))}
-      <SocialShareIcon
-        href={getTwitterShareLink(link, post?.title)}
-        icon={<TwitterIcon />}
-        className="bg-theme-bg-twitter"
-        onClick={() => trackClick(ShareProvider.Twitter)}
-        label="Twitter"
-      />
-      <SocialShareIcon
-        href={getWhatsappShareLink(link)}
-        icon={<WhatsappIcon />}
-        onClick={() => trackClick(ShareProvider.WhatsApp)}
-        className="bg-theme-bg-whatsapp"
-        label="WhatsApp"
-      />
-      <SocialShareIcon
-        href={getFacebookShareLink(link)}
-        icon={<FacebookIcon />}
-        className="bg-theme-bg-facebook"
-        onClick={() => trackClick(ShareProvider.Facebook)}
-        label="Facebook"
-      />
-      <SocialShareIcon
-        href={getRedditShareLink(link, post?.title)}
-        icon={<RedditIcon />}
-        className="bg-theme-bg-reddit"
-        onClick={() => trackClick(ShareProvider.Reddit)}
-        label="Reddit"
-      />
-      <SocialShareIcon
-        href={getLinkedInShareLink(link)}
-        icon={<LinkedInIcon />}
-        className="bg-theme-bg-linkedin"
-        onClick={() => trackClick(ShareProvider.LinkedIn)}
-        label="LinkedIn"
-      />
-      <SocialShareIcon
-        href={getTelegramShareLink(link, post?.title)}
-        icon={<TelegramIcon />}
-        className="bg-theme-bg-telegram"
-        onClick={() => trackClick(ShareProvider.Telegram)}
-        label="Telegram"
-      />
-      <SocialShareIcon
-        href={getEmailShareLink(link, 'I found this amazing post')}
-        icon={<MailIcon />}
-        className="bg-theme-bg-email"
-        onClick={() => trackClick(ShareProvider.Email)}
-        label="Email"
-      />
-    </section>
+    <>
+      {hasSquadAccess &&
+        type === SocialShareType.Squad &&
+        !isComment &&
+        !post.private && (
+          <>
+            <p className="py-2.5 font-bold typo-callout">
+              Share with your squad
+            </p>
+            <section className="grid grid-cols-5 gap-4 pt-2 w-fit">
+              {squads
+                ?.filter(
+                  (squadItem) =>
+                    squadItem.active &&
+                    verifyPermission(squadItem, SourcePermissions.Post),
+                )
+                ?.map((squad) => (
+                  <button
+                    type="button"
+                    className="flex flex-col items-center w-16 text-center"
+                    key={squad.id}
+                    onClick={() => onShareToSquad(squad)}
+                  >
+                    <SourceProfilePicture source={squad} />
+                    <ShareText className="mt-2 break-words">
+                      @{squad.handle}
```

Comment:
Did you take the new guideline in sketch into account, it should have a max-width and dots when overflowing.

![Uploading Screenshot 2023-06-14 at 13.49.34.png…]()